src/spikeinterface/preprocessing/deepinterpolation/generators.py

```diff
@@ -86,11 +86,11 @@ def __init__(
         sequential_generator_params["total_samples"] = self.total_samples
         sequential_generator_params["pre_post_omission"] = pre_post_omission
 
-        json_fd, json_path = tempfile.mkstemp(suffix=".json")
-        os.close(json_fd)
-        with open(json_path, "w") as f:
+        with tempfile.NamedTemporaryFile(suffix=".json") as f:
             json.dump(sequential_generator_params, f)
```

Comment:
This seems to be failing...

Reply:
New error I have not idea. Any suggestions? It seems is not finding the training weights?

Reply:
It looks like it's the same problem no? Still failing at dump (check the first failure, all later ones are dependant on it)

Reply:
Yeah, you are right, I was confused by the rest of the stack trace. It was just old binary mode vs text mode in the context manager.